CONFIG_BT_MESH_ADV_EXT doesn't build without CONFIG_BT_MESH_RELAY

**Describe the bug**

Bluetooth Mesh using Extended advertising seems to have an incorrect compile-time dependency on  relay support.

**To Reproduce**
Steps to reproduce the behavior:
```
west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world \
    -DCONFIG_BT=y \
    -DCONFIG_BT_BROADCASTER=y \
    -DCONFIG_BT_OBSERVER=y \
    -DCONFIG_BT_MESH=y \
    -DCONFIG_BT_EXT_ADV=y \
    -DCONFIG_BT_CTLR_ADV_EXT=y
```
The above command results in the following compiler error on the v3.0.0 tag:

```
/tmp/zephyrproject/zephyr/subsys/bluetooth/mesh/adv_ext.c:83:34: error: 'CONFIG_BT_MESH_RELAY_ADV_SETS' undeclared here (not in a function); did you mean 'CONFIG_BT_MESH_ADV_EXT'?
   83 | #define BT_MESH_ADV_COUNT   (1 + CONFIG_BT_MESH_RELAY_ADV_SETS)
```

--- a/subsys/bluetooth/mesh/adv_ext.c
+++ b/subsys/bluetooth/mesh/adv_ext.c
@@ -26,6 +26,10 @@
 /* Convert from ms to 0.625ms units */
 #define ADV_INT_FAST_MS    20
 
+#ifndef CONFIG_BT_MESH_RELAY_ADV_SETS
+#define CONFIG_BT_MESH_RELAY_ADV_SETS 0
+#endif
+
 enum {
 	/** Controller is currently advertising */
 	ADV_FLAG_ACTIVE,

